### PR TITLE
MultiSelect Composite

### PR DESCRIPTION
**Pull request by Wo1v3r**
### What changed
Added Composite MultiSelect 
...

### Why it changed
Because it is required for Smart Actions.
We should invest the time when possible in a more general way of composing input + label as it is a very common requirement.
...

---

- [ X] Change is tested
- [X ] Change is documented
- [ ] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

**Comment by lbelinsk** [comment]
@Wo1v3r We would prefer having one generic composite instead of various ad-hoc ones, but since you created it I assume you need it so we probably shouldn't slow you down

File: testkit/enzyme.js [hunk 1]
```diff
@@ -78,6 +78,9 @@ export const dropdownCompositeTestkitFactory = enzymeTestkitFactoryCreator(dropd
 import autoCompleteCompositeDriverFactory from '../src/AutoCompleteComposite/AutoCompleteComposite.driver';
 export const autoCompleteCompositeTestkitFactory = enzymeTestkitFactoryCreator(autoCompleteCompositeDriverFactory);
 
+import multiSelectCompositeDriverFactory from '../src/MultiSelectComposite/MultiSelectComposite.driver';
+export const multiSelectCompositeTestkitFactory = enzymeTestkitFactoryCreator(multiSelectCompositeDriverFactory);
```
Comment (lbelinsk) [inline testkit/enzyme.js]: What about the 3r'd testkit (the `testkit/index.js` file)

File: stories/MultiSelectComposite/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {storiesOf} from '@storybook/react';
+import InteractiveCodeExample from 'wix-storybook-utils/InteractiveCodeExample';
+import Markdown from 'wix-storybook-utils/Markdown';
+
+import Readme from '../../src/MultiSelectComposite/README.md';
+import ExampleStandard from './ExampleStandard';
+
+storiesOf('3. Inputs', module).add('3.8 + MultiSelectComposite', () => (
```
Comment (lbelinsk) [inline stories/MultiSelectComposite/index.js]: We already have 3.8. @argshook what do you say?

Comment (argshook) [reply stories/MultiSelectComposite/index.js]: well i would prefer to not add this component at all and instead focus on more generic way for label + input, as @Wo1v3r said.

if it's not what we're gonna do, then 3.8 + MultiSelectComposite` i guess is okay

Comment (Wo1v3r) [reply stories/MultiSelectComposite/index.js]: When we'll cool down with Smart Actions I can PR a general Composer, I'd love to pr one based on component types or maybe one that takes an order of arranging the components.
We do need it for the end of the week so let me know if there's anything additional to promote this current  PR ^_^

File: testkit/enzyme.js (hunk 1 above)
Comment (Wo1v3r) [reply testkit/enzyme.js]: I'll add and tag for review ^^

**Comment by lbelinsk** [comment]
I suggest to merge this, and then to remove it after we will have the generic composite..

**Comment by hepiyellow** [comment]
@lbelinsk @argshook 
 I think we should have separately a generic CompositeLabel, and a generic CompositeContainer.

For a CompositeLabel, we might want to create a component called <LabeledField> (or even use the existing <Label>) which could receiving **children or any kind.**


The CompositeContainer is required only in components like <Range> and <FieldWithSelection> which (for the moment) will support only HBox layout of 2 items, and will manage the borders and focus borders behaviour. 

We don't have to implement this CompositeContainer now. I suggest we first release a breaking change which  separates components like <DropdownComposite>  into <LabeledField><Dropdown/></LabeledField>. 

What say you?

**Pull request merged**